app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -0,0 +1,24 @@
+package ru.yandex.practicum.contacts.presentation.base;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.DiffUtil;
+
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
+
+    @Override
+    public boolean areItemsTheSame(@NonNull T oldItem, @NonNull T newItem) {
+        return oldItem.theSameAs(oldItem) == newItem.theSameAs(newItem);
```

Comment:
```suggestion
        return oldItem.theSameAs(newItem);
```
Сейчас в твоей реализации функция всегда будет возвращать `false`, т.к. старый айтем всегда равен старому айтему (левый операнд) и новый айтем всегда равен новому айтему (правый операнд), собственно true == true будет всегда true. Здесь же нужно проверить, является ли новый айтем "тем же", что и старый айтем

Reply:
Согласен, с обеих сторон всегда будет true, обрадовавшись, что нашел решение, толком не проверил его логику. Исправил в коммите 9599fa76473a94cf1c0a537bfe563a614ae62813